Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewCell.cs

```diff
@@ -0,0 +1,1265 @@
+        public Rectangle GetContentBounds(int rowIndex)
+        {
+            Contract.Requires(this.DataGridView != null);
+            Contract.Requires(rowIndex >= 0 || rowIndex < this.DataGridView.Rows.Count);
+            return default(Rectangle);
+        }
+
+        // <summary>
+        // Returns the bounding rectangle that encloses the cell's content area, which is calculated using the specified <see cref="T:System.Drawing.Graphics"/> and cell style.
+        // </summary>
+        // 
+        // <returns>
+        // The <see cref="T:System.Drawing.Rectangle"/> that bounds the cell's contents.
+        // </returns>
+        // <param name="graphics">The graphics context for the cell.</param><param name="cellStyle">The <see cref="T:System.Windows.Forms.DataGridViewCellStyle"/> to be applied to the cell.</param><param name="rowIndex">The index of the cell's parent row.</param>
+        // protected virtual Rectangle GetContentBounds(Graphics graphics, DataGridViewCellStyle cellStyle, int rowIndex)
+        
+        /// <summary>
+        /// Returns the current, formatted value of the cell, regardless of whether the cell is in edit mode and the value has not been committed.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The current, formatted value of the <see cref="T:System.Windows.Forms.DataGridViewCell"/>.
+        /// </returns>
+        /// <param name="rowIndex">The row index of the cell.</param><param name="context">A bitwise combination of <see cref="T:System.Windows.Forms.DataGridViewDataErrorContexts"/> values that specifies the data error context.</param><exception cref="T:System.ArgumentOutOfRangeException">The specified <paramref name="rowIndex"/> is less than 0 or greater than the number of rows in the control minus 1. </exception><exception cref="T:System.InvalidOperationException"><see cref="P:System.Windows.Forms.DataGridViewCell.ColumnIndex"/> is less than 0, indicating that the cell is a row header cell.</exception><exception cref="T:System.Exception">Formatting failed and either there is no handler for the <see cref="E:System.Windows.Forms.DataGridView.DataError"/> event of the <see cref="T:System.Windows.Forms.DataGridView"/> control or the handler set the <see cref="P:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException"/> property to true. The exception object can typically be cast to type <see cref="T:System.FormatException"/>.</exception>
+        public object GetEditedFormattedValue(int rowIndex, DataGridViewDataErrorContexts context)
+        {
+            Contract.Requires(this.DataGridView != null);
```

Comment:
The same is true here. I don't the this checks in the source code.

Here what I see:

```
        public object GetEditedFormattedValue(int rowIndex, DataGridViewDataErrorContexts context)
        {
            if (this.DataGridView == null)
            {
                return null;
            }
            DataGridViewCellStyle dataGridViewCellStyle = GetInheritedStyle(null, rowIndex, false /*includeColors*/);
            return GetEditedFormattedValue(GetValue(rowIndex), rowIndex, ref dataGridViewCellStyle, context);
        }
```